src/components/InputProblem.tsx

```diff
@@ -105,11 +121,54 @@ export const InputProblem = ({
     return trimmedInput.toLowerCase() === trimmedAnswer.toLowerCase()
   }
 
+  const setPersistedState = async (): Promise<void> => {
```

Comment:
I think this can be pulled out and moved into a helper function in `handleSubmit()`.